10-inference.Rmd

```diff
@@ -0,0 +1,789 @@
+Vancouver, Canada. We will draw a single sample of size 40 from the
+population and visualize the distribution of the sample:
+
+```{r 11-bootstrapping1, echo = TRUE, message = FALSE, warning = FALSE, fig.cap = "Histogram of price per night ($) for one sample of size 40"}
+one_sample <- airbnb %>% 
+    rep_sample_n(40) %>% 
+    ungroup() %>% # ungroup the data frame 
+    select(price) # drop the replicate column 
+head(one_sample)
+one_sample_dist <- ggplot(one_sample, aes(price)) + 
+    geom_histogram(fill="#0072B2", color="#e9ecef") +
+    xlab("Price per night ($)") 
+one_sample_dist
+```
+
+```{r 11-bootstrapping2, echo = TRUE, message = FALSE, warning = FALSE}
+one_sample_estimates <- one_sample %>% 
+    summarise(sample_mean = mean(price))
+one_sample_estimates
+```
+
+The sample distribution is skewed with a few observations out to the right. The
+mean of the sample is \$`r round(one_sample_estimates$sample_mean,2)`.
+Remember, in practice, we usually only have one sample from the population. So
+this sample and estimate are the only data we can work with.
+
+We now perform steps 1 - 5 listed above to generate a single bootstrap sample in R using the
+sample we just took, and calculate the bootstrap estimate for that sample. We
+will use the `rep_sample_n` function as we did when we were creating our
+sampling distribution. Since we want to sample with replacement, we change the
+argument for `replace` from its default value of `FALSE` to `TRUE`.
+
+```{r 11-bootstrapping3, echo = TRUE, message = FALSE, warning = FALSE, fig.cap = "Bootstrap distribution"}
+boot1 <- one_sample %>%
+    rep_sample_n(size = 40, replace = TRUE, reps = 1)
+head(boot1)
+boot1_dist <- ggplot(boot1, aes(price)) + 
+    geom_histogram(fill="#0072B2", color="#e9ecef") +
+    xlab("Price per night ($)") 
+
+boot1_dist
+
+summarise(boot1, mean = mean(price))
+```
+
+Notice that our bootstrap distribution has a similar shape to the original
+sample distribution. Though the shapes of the distributions are similar, they
+are not identical. You'll also notice that the original sample mean and the
+bootstrap sample mean differ. How might that happen? Remember that we are
+sampling with replacement from the original sample, so we don't end up with the
+same sample values again. We are trying to mimic drawing another sample from
+the population without actually having to do that.
+
+Let's now take 1500 bootstrap samples from the original sample we drew from the
+population (`one_sample`)  using `rep_sample_n` and calculate the means for
+each of those replicates. Recall that this assumes that `one_sample` *looks like*
+our original population; but since we do not have access to the population itself,
+this is often the best we can do.
+
+```{r 11-bootstrapping4, echo = TRUE, message = FALSE, warning = FALSE}
+boot1500 <- one_sample %>%
+    rep_sample_n(size = 40, replace = TRUE, reps = 1500)
+head(boot1500)
+tail(boot1500)
```

Comment:
After showing `head` and `tail`, it might be nice to show 6 or so sample distributions to show how they differ, something like this: 

<img width="525" alt="Screen Shot 2020-11-19 at 12 59 34 PM" src="https://user-images.githubusercontent.com/6865016/99723508-22df7500-2a67-11eb-808f-eb9543b9b7cb.png">

And then calculate and show the point estimates for these six, or so, samples:

<img width="538" alt="Screen Shot 2020-11-19 at 12 59 48 PM" src="https://user-images.githubusercontent.com/6865016/99723555-3094fa80-2a67-11eb-8ebd-1ca698e1e4ba.png">

Reply:
Source for this is here: https://pages.github.ubc.ca/MDS-2020-21/DSCI_552_stat-inf-1_students/lectures/02_lecture-sampling-and-bootstrapping.html